src/sentry/static/sentry/app/components/globalSelectionLink.jsx

```diff
@@ -0,0 +1,52 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+import {Link as RouterLink} from 'react-router';
+import {pick} from 'lodash';
+
+import {URL_PARAM} from 'app/components/organizations/globalSelectionHeader/constants';
+
+/**
+ * A modified link used for navigating between organization level pages that
+ * will keep the global selection values (projects, environments, time) in the
+ * querystring when navigating if it's present
+ *
+ * Falls back to <a> if there is no router present.
+ */
+export default class GlobalSelectionLink extends React.Component {
+  static propTypes = {
+    to: PropTypes.oneOfType([PropTypes.string, PropTypes.object]).isRequired,
+  };
+
+  static contextTypes = {
+    location: PropTypes.object,
+  };
+
+  render() {
+    const {location} = this.context;
+
+    const query = pick(location.query, Object.values(URL_PARAM));
+
+    if (location) {
+      const hasQuery = Object.keys(query).length > 0;
+
+      let {to} = this.props;
+
+      if (hasQuery) {
+        if (typeof to === 'string') {
+          to = {pathname: to, query};
+        }
+      }
+
+      const routerProps = to ? {...this.props, to} : {...this.props};
+
+      return <RouterLink {...routerProps}>{this.props.children}</RouterLink>;
+    } else {
+      let {to, ...props} = this.props;
+      return (
+        <a {...props} href={to}>
```

Comment:
In the fallback case could you append the serialized querystring?

Reply:
How do you think we should get this? (since in the fallback we won't have access to `this.context.location.search`). I also don't think this situation is likely in scenarios where we'll need to preserve the querystring.

Reply:
Yeah, there isn't a good place to get the querystring if there is no `location`.